src/filters/std/array.rs

```diff
@@ -4,9 +4,20 @@ use liquid_derive::*;
 use liquid_error::Result;
 use liquid_interpreter::Context;
 use liquid_interpreter::Expression;
-use liquid_value::{Object, Scalar, Value};
+use liquid_value::{Scalar, Value};
 use std::cmp;
 
+macro_rules! as_sequence {
```

Comment:
I wonder if this would be worthwhile to turn into an iterator for `Value`

Reply:
When I thought about how to implement it, I could have gone with one of the following ways:
* Using a boxed dynamic iterator
* Creating an enum of `slice::Iter`, `iter:Once` and `iter::Empty`
* Using a generic function.

I tried to avoid the first two because of the runtime cost (even the enum would have to query itself every invocation of `next`), but I also didn't like the syntactic overhead of the third option. The macro was a nice compromise (at least I think so).

I, since Ruby used `InputIterator` only for filters, I hesitant about embedding this into `Value`.

Reply:
If its just a couple of places then that might be fine.  I tried to centralize general properties.

I'm not too worry about the `Box` overhead.  The `clone`s far outweigh any difference that the `Box` would have.